### PR TITLE
Replace deprecated set-output in GHA

### PR DESCRIPTION
**Pull request by atugushev**
https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,14 +71,10 @@ jobs:
...
         run: python --version --version
-      - name: Get pip cache dir
-        id: pip-cache
-        run: |
-          echo "::set-output name=dir::$(pip cache dir)"
       - name: Pip cache
         uses: actions/cache@v3
         with:
-          path: ${{ steps.pip-cache.outputs.dir }}
+          path: $(pip cache dir)
```
Comment (webknjaz) [inline .github/workflows/ci.yml:77]: You can't do shell invocations right in the action inputs. It must be in a run-step, setting a var that can then be rendered. GHA only interpolates own syntax, not bash.

File: .github/workflows/qa.yml [hunk 2]
```diff
@@ -51,7 +47,7 @@ jobs:
             ${{ runner.os }}-
       - name: Prepare cache key
         id: cache-key
-        run: echo "::set-output name=sha-256::$(python -VV | sha256sum | cut -d' ' -f1)"
+        run: echo "sha-256=$(python -VV | sha256sum | cut -d' ' -f1)" >> $GITHUB_OUTPUT
```
Comment (webknjaz) [inline .github/workflows/qa.yml:50]: This change is the only one correct in the PR.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (atugushev) [reply .github/workflows/ci.yml]: You're supposed not to see this ;D

Comment (webknjaz) [reply .github/workflows/ci.yml]: _/me sees everything_

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
GHA is drunk these days. I've restarted the windows jobs.

**Pull request merged**